api/src/main/scala/com/lightbend/lagom/internal/api/Path.scala

```diff
@@ -7,57 +7,55 @@ import java.util.regex.Pattern
 
 import akka.util.ByteString
 import com.lightbend.lagom.javadsl.api.Descriptor.{ CallId, NamedCallId, PathCallId, RestCallId }
-import com.lightbend.lagom.javadsl.api.deser.RawId.PathParam
-import com.lightbend.lagom.javadsl.api.deser.{ RawId, RawIdDescriptor }
-import org.pcollections.{ PSequence, PMap, HashTreePMap, TreePVector }
 import play.utils.UriEncoding
 
-import scala.collection.JavaConverters._
 import scala.util.parsing.combinator.JavaTokenParsers
 
-case class Path(parts: Seq[PathPart], queryParams: Seq[String]) {
+case class Path(pathSpec: String, parts: Seq[PathPart], queryParams: Seq[String]) {
```

Comment:
I've never agreed with this.  Yes `Seq` *can* be mutable.  Yes a malicious developer could destroy your program by exploiting that.  But this is an internal API, and the people working on Lagom are not malicious.  I don't think the additional complexity added to code to use `immutable.Seq` everywhere is worth that pain.